### PR TITLE
Try launching in external app first

### LINKED ISSUE #42: PlatformException after opening maps on IOS
When using `MapsLauncher.launchQuery()` the Maps app opens correctly with the position but in background a PlatformException is triggered. 
The throwing method is _failedSafariViewControllerLoadException() of url_launcher_ios package.

Versions:
maps_launcher: 2.2.1
url_launcher: 6.2.6
url_launcher_ios: 6.2.5

### PR DESCRIPTION
**Pull request by lukehutch**
Follows the advice [here](https://stackoverflow.com/a/74012029/3950982) to try to fix #42.

(The changes are all in maps_launcher.dart, but I also updated the deps, hence the large changeset)

### CODE REVIEW

**Comment by mccadi** [comment]
Thank you, hopefully this gets a look from the owner soon.

**Comment by edwin-alvarez** [comment]
@pikaju , can this be merged?

**Comment by saibotma** [comment]
This should also fix https://github.com/pikaju/flutter-maps-launcher/issues/38 and make https://github.com/pikaju/flutter-maps-launcher/pull/39 obsolete.

File: lib/maps_launcher.dart [hunk 1]
```diff
@@ -56,20 +56,33 @@ class MapsLauncher {
     return uri;
   }
 
+  static Future<LaunchMode> _launchMode() async =>
+      await supportsLaunchMode(LaunchMode.externalNonBrowserApplication)
```
Comment (LevisLuong) [inline lib/maps_launcher.dart:60]: @lukehutch  Please help me check if the method `supportsLaunchMode` exists. I received this error when trying to build
Error (Xcode): ../../../../.pub-cache/git/flutter-maps-launcher-470dd6965ea83de929851c48723f5745dbafbfcd/lib/maps_launcher.dart:60:13: Error: Method not found: 'supportsLaunchMode'.

Comment (lukehutch) [reply lib/maps_launcher.dart:60]: Yes, this method is in `url_launcher`, which is a dependency of `flutter_maps_launcher`.

Comment (lukehutch) [reply lib/maps_launcher.dart:60]: I did update the deps versions though, as I said in the description, so make sure you update those, then dart `pub upgrade`

**Comment by pikaju** [comment]
Hey, I had merge conflicts with the other PRs, but I integrated your idea with the `_launchMode` function now. Thank you!

**Pull request closed without merging**